### PR TITLE
feat: add CLI pull command 

### PR DESCRIPTION
**Pull request by enricoschuetz**
Changes:

+ added CLI `pull` command to fetch and save multiple Thing Models
+ added name pattern filter as argument to CLI `list` command

### CODE REVIEW

File: internal/model/search.go [hunk 1]
```diff
@@ -128,4 +133,11 @@ type SearchParams struct {
 	ExternalID   []string
 	Name         string
 	Query        string
+	Options      *SearchOptions
```
Comment (alexbrdn) [inline internal/model/search.go:136]: why `*SearchOptions`, as opposed to `SearchOptions`?

File: internal/model/search_test.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package model
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSearch_FilterType(t *testing.T) {
```
Comment (alexbrdn) [inline internal/model/search_test.go:9]: What's the purpose of this test? I'm puzzled

Comment (enricoschuetz) [reply internal/model/search_test.go:9]: The intention was to have an additional , lets say "safe guard" for developers, not easy to change the order of the enum and give them a hint.
When you create an empty SearchOptions `... := SearchOptions{}`, of course the value zero is the default value for `NameFilterType`.  
If the enum iota order is changed (maybe by some mistake) and the app behavior is not correctly covered by other tests (e.g. by not exact counting of the search result), we could have easily a bug that you dont see on the first view.

File: cmd/pull.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+		os.Exit(1)
+	}
+
+	name := ""
+	if len(args) > 0 {
+		name = args[0]
+	}
+	search := cli.CreateSearchParamsFromCLI(pFilterFlags, name)
+	if search != nil {
+		search.Options = &model.SearchOptions{NameFilterType: model.PrefixMatch}
```
[the remark this answers is not in the document]
Comment (enricoschuetz) [reply cmd/pull.go:55]: It was just for better understanding what the expression of the default value (behavior) is.
We can also remove it and set `PrefixMatch = iota +1`,

**Review by alexbrdn — APPROVED** [review]
👍

**Pull request merged**